### PR TITLE
Kube/OIDC

### PR DESCRIPTION
**Pull request by matprig**
Hello,

This PR contains:
- kube OIDC parameter resource
- documentation
- test acceptance
best

### CODE REVIEW

**Review by XavierDuthil — CHANGES_REQUESTED** [review]

File: ovh/resource_cloud_project_kube_oidc.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	endpoint := fmt.Sprintf("/cloud/project/%s/kube/%s/openIdConnect", serviceName, kubeID)
+	res := &CloudProjectKubeOIDCResponse{}
+
+	log.Printf("[DEBUG] Will read oidc from kube %s and project: %s", kubeID, serviceName)
+	err := config.OVHClient.Get(endpoint, res)
+	if err != nil {
+		return fmt.Errorf("calling get %s %q", endpoint, err)
+	}
+	for k, v := range res.ToMap() {
+		if k != "id" {
```
Comment (XavierDuthil) [inline ovh/resource_cloud_project_kube_oidc.go:83]: I would have inverted the conditions

**Review by XavierDuthil — CHANGES_REQUESTED** [review]

File: ovh/resource_cloud_project_kube_oidc.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"issuer_url": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
```
[the remark this answers is not in the document]
Comment (matprig) [reply ovh/resource_cloud_project_kube_oidc.go:39]: - [ ] true need to do gofumtp on all files on all PRs

File: ovh/resource_cloud_project_kube_oidc.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+	kubeID := d.Get("kube_id").(string)
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/kube/%s/openIdConnect", serviceName, kubeID)
+	params := (&CloudProjectKubeOIDCCreateOpts{}).FromResource(d)
+	res := &CloudProjectKubeOIDCResponse{}
+
+	log.Printf("[DEBUG] Will create OIDC: %+v", params)
+	err := config.OVHClient.Post(endpoint, params, res)
+	if err != nil {
+		return fmt.Errorf("calling Post %s with params %s:\n\t %q", endpoint, params, err)
```
[the remark this answers is not in the document]
Comment (matprig) [reply ovh/resource_cloud_project_kube_oidc.go:54]: - [ ] true this is a point, i did not notice that the whole project was not wrapping errors %w directive

File: ovh/resource_cloud_project_kube_oidc.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (XavierDuthil) [reply ovh/resource_cloud_project_kube_oidc.go:39]: /approve

File: website/docs/r/cloud_project_kube_oidc.html.markdown [hunk 4]
```diff
@@ -0,0 +1,35 @@
+---
+layout: "ovh"
+page_title: "OVH: cloud_project_kube_oidc"
+sidebar_current: "docs-ovh-resource-cloud-project-kube-oidc"
+description: |-
+Creates an oidc configuration in a kubernetes managed cluster.
```
Comment (scraly) [inline website/docs/r/cloud_project_kube_oidc.html.markdown:6]:
```suggestion
Creates an OIDC configuration in an OVHcloud Managed Kubernetes cluster.
```

File: website/docs/r/cloud_project_kube_oidc.html.markdown [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+---
+
+# ovh_cloud_project_kube_oidc
+
+Creates a oidc configuration in a OVH Managed Kubernetes Service cluster.
```
Comment (scraly) [inline website/docs/r/cloud_project_kube_oidc.html.markdown:11]:
```suggestion
Creates an OIDC configuration in an OVHcloud Managed Kubernetes cluster.
```

File: website/docs/r/cloud_project_kube_oidc.html.markdown [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  client_id    = "xxx"
+  issuer_url   = "https://ovh.com"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `service_name` - (required) The id of the public cloud project. If omitted,
```
Comment (scraly) [inline website/docs/r/cloud_project_kube_oidc.html.markdown:28]:
```suggestion
* `service_name` - (required) The ID of the public cloud project. If omitted,
```

File: website/docs/r/cloud_project_kube_oidc.html.markdown [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  the `OVH_CLOUD_PROJECT_SERVICE` environment variable is used.
+
+* `kube_id` - The id of the managed kubernetes cluster.
```
Comment (scraly) [inline website/docs/r/cloud_project_kube_oidc.html.markdown:31]:
```suggestion
* `kube_id` - The ID of the managed kubernetes cluster.
```

File: website/docs/r/cloud_project_kube_oidc.html.markdown [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+# ovh_cloud_project_kube_oidc
+
+Creates a oidc configuration in a OVH Managed Kubernetes Service cluster.
+
+## Example Usage
+
+```hcl
+resource "ovh_cloud_project_kube_oidc" "my-oidc" {
+  service_name = var.projectid
+  kube_id      = ovh_cloud_project_kube.k8stf.id
```
Comment (scraly) [inline website/docs/r/cloud_project_kube_oidc.html.markdown:18]:
```suggestion
  kube_id      = ovh_cloud_project_kube. mykube.id
```

File: ovh/types_cloud.go [hunk 9]
```diff
@@ -117,7 +117,7 @@ func (u CloudProjectUser) ToMap() map[string]interface{} {
 	obj := make(map[string]interface{})
 	obj["creation_date"] = u.CreationDate
 	obj["description"] = u.Description
-	//Dont set password as it must be set only at creation time
+	// Dont set password as it must be set only at creation time
```
Comment (scraly) [inline ovh/types_cloud.go:120]:
```suggestion
	// Don't set password as it must be set only at creation time
```

**Review by scraly — CHANGES_REQUESTED** [review]

**Comment by scraly** [comment]
OIDC provider

🙂

File: ovh/resource_cloud_project_kube_oidc_test.go [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+		"xxx",
+		"https://ovh.com",
+	)
+
+	configUpdated := fmt.Sprintf(
+		testAccCloudProjectKubeOIDCConfig,
+		os.Getenv("OVH_CLOUD_PROJECT_SERVICE_TEST"),
+		name,
+		region,
+		"yyy",
```
Comment (scraly) [inline ovh/resource_cloud_project_kube_oidc_test.go:44]:
```suggestion
		"my-another-oidc-client-id",
```

File: ovh/resource_cloud_project_kube_oidc_test.go [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+func TestAccCloudProjectKubeOIDC_full(t *testing.T) {
+	region := os.Getenv("OVH_CLOUD_PROJECT_KUBE_REGION_TEST")
+
+	name := acctest.RandomWithPrefix(test_prefix)
+	config := fmt.Sprintf(
+		testAccCloudProjectKubeOIDCConfig,
+		os.Getenv("OVH_CLOUD_PROJECT_SERVICE_TEST"),
+		name,
+		region,
+		"xxx",
```
Comment (scraly) [inline ovh/resource_cloud_project_kube_oidc_test.go:35]:
```suggestion
		"my-oidc-client-id",
```

File: ovh/resource_cloud_project_kube_oidc_test.go [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+						"ovh_cloud_project_kube_oidc.my-oidc", "client_id", "xxx"),
+					resource.TestCheckResourceAttr(
+						"ovh_cloud_project_kube_oidc.my-oidc", "issuer_url", "https://ovh.com"),
+				),
+			},
+			{
+				Config: configUpdated,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"ovh_cloud_project_kube_oidc.my-oidc", "client_id", "yyy"),
```
Comment (scraly) [inline ovh/resource_cloud_project_kube_oidc_test.go:69]:
```suggestion
						"ovh_cloud_project_kube_oidc.my-oidc", "client_id", "my-another-oidc-client-id"),
```

File: ovh/resource_cloud_project_kube_oidc_test.go [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+			testAccCheckCloudProjectExists(t)
+			testAccPreCheckKubernetes(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"ovh_cloud_project_kube_oidc.my-oidc", "client_id", "xxx"),
```
Comment (scraly) [inline ovh/resource_cloud_project_kube_oidc_test.go:60]:
```suggestion
						"ovh_cloud_project_kube_oidc.my-oidc", "client_id", "my-oidc-client-id"),
```

**Review by scraly — CHANGES_REQUESTED** [review]

File: website/docs/r/cloud_project_kube_oidc.html.markdown [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+# ovh_cloud_project_kube_oidc
+
+Creates an OIDC configuration in an OVHcloud Managed Kubernetes cluster.
+
+## Example Usage
+
+```hcl
+resource "ovh_cloud_project_kube_oidc" "my-oidc" {
+  service_name = var.projectid
+  kube_id      = ovh_cloud_project_kube. mykube.id
```
Comment (scraly) [inline website/docs/r/cloud_project_kube_oidc.html.markdown:18]:
```suggestion
  kube_id      = ovh_cloud_project_kube.mykube.id
```

**Review by scraly — CHANGES_REQUESTED** [review]

**Review by scraly — APPROVED** [review]

**Pull request merged**